api/src/test/java/com/spotify/ffwd/util/HighFrequencyDetectorTest.java

```diff
@@ -0,0 +1,182 @@
+import com.spotify.ffwd.statistics.HighFrequencyDetectorStatistics;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@RunWith(MockitoJUnitRunner.class)
+public class HighFrequencyDetectorTest {
+
+  private final boolean dropHighFrequencyMetric = true;
+  private final int minFrequencyMillisAllowed = 1000;
+  private final long highFrequencyDataRecycleMS = 300_000;
+  private final int minNumberOfTriggers = 2;
+
+  private HighFrequencyDetector detector;
+  private HighFrequencyDetector shortSwapDetector;
+  private Logger log;
+
+  @Mock private HighFrequencyDetectorStatistics statistics;
+
+  @Before
+  public void setup() {
+    log = LoggerFactory.getLogger(getClass());
+    detector = createHighFreqDetector(highFrequencyDataRecycleMS);
+    shortSwapDetector = createHighFreqDetector(500L);
+  }
+
+  public HighFrequencyDetector createHighFreqDetector(long swapMS) {
+    final List<Module> modules = Lists.newArrayList();
+
+    modules.add(
+        new AbstractModule() {
+          @Override
+          protected void configure() {
+            bind(Logger.class).toInstance(log);
+            bind(Boolean.class)
+                .annotatedWith(Names.named("dropHighFrequencyMetric"))
+                .toInstance(dropHighFrequencyMetric);
+            bind(Integer.class)
+                .annotatedWith(Names.named("minFrequencyMillisAllowed"))
+                .toInstance(minFrequencyMillisAllowed);
+            bind(Long.class)
+                .annotatedWith(Names.named("highFrequencyDataRecycleMS"))
+                .toInstance(swapMS);
+            bind(Integer.class)
+                .annotatedWith(Names.named("minNumberOfTriggers"))
+                .toInstance(minNumberOfTriggers);
+            bind(HighFrequencyDetectorStatistics.class).toInstance(statistics);
+          }
+        });
+
+    final Injector injector = Guice.createInjector(modules);
+
+    return injector.getInstance(HighFrequencyDetector.class);
+  }
+
+  @Test
+  public void testHighFrequencyDetection() {
+    // verifies if high frequency metrics are captured and sortable
+    List<Metric> finalList = new ArrayList<>();
+    for (int y = 0; y < 3; y++) {
+      for (int x = 0; x < 3; x++) {
+        List<Metric> list = new ArrayList<>();
+        for (int i = 0; i < 30; i++) {
+          list.add(
+              new Metric(
+                  "KEY" + y,
+                  42.0 + i,
+                  new Date(),
+                  ImmutableSet.of(),
+                  Map.of("tag1", "value1", "what", "fun"),
+                  ImmutableMap.of(),
+                  null));
+        }
+
+        finalList.addAll(detector.detect(list));
+      }
+    }
+    assertEquals(90, finalList.size());
```

Comment:
should we be testing that it's sorted properly?

Reply:
I don't care about sorting itself as long as it is consistent.